flink-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -392,6 +425,45 @@ public void testCassandraPojoAtLeastOnceSink() throws Exception {
 	}
 
 	@Test
+	public void testCassandraTableSink() throws Exception {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		env.setParallelism(1);
+
+		DataStreamSource<Row> source = env.fromCollection(rowCollection);
+		CassandraTableSink cassandraTableSink = new CassandraTableSink(new ClusterBuilder() {
+			@Override
+			protected Cluster buildCluster(Cluster.Builder builder) {
+				return builder.addContactPointsWithPorts(new InetSocketAddress(HOST, PORT)).build();
+			}
+		}, injectTableName(INSERT_DATA_QUERY), new Properties());
+		CassandraTableSink newCassandrTableSink = cassandraTableSink.configure(FIELD_NAMES, FIELD_TYPES);
+
+		newCassandrTableSink.emitDataStream(source);
+
+		env.execute();
+		ResultSet rs = session.execute(injectTableName(SELECT_DATA_QUERY));
+		Assert.assertEquals(20, rs.all().size());
+	}
+
+	@Test
+	public void testCassandraTableSinkE2E() throws Exception {
```

Comment:
What's the difference between this test and `TestCassandraTableSink`?

Reply:
this is added as another reviewer think better to add e2e test using sql api.

Reply:
I think `testCassandraTableSink` can be removed in favor of this test.